specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/preview/2024-01-01-preview/workspaceRP.json

```diff
@@ -1798,6 +2706,122 @@
       },
       "x-ms-discriminator-value": "ApiKey"
     },
+    "CognitiveServiceEndpointDeploymentResourceProperties": {
+      "required": [
+        "model"
+      ],
+      "type": "object",
+      "properties": {
+        "model": {
+          "description": "Model used for the endpoint deployment.",
+          "$ref": "#/definitions/EndpointDeploymentModel"
+        },
+        "properties": {},
+        "raiPolicyName": {
+          "type": "string"
+        },
+        "sku": {
+          "$ref": "#/definitions/CognitiveServicesSku"
+        },
+        "versionUpgradeOption": {
+          "$ref": "#/definitions/DeploymentModelVersionUpgradeOption"
+        }
+      }
+    },
+    "CognitiveServicesSku": {
+      "type": "object",
+      "properties": {
+        "capacity": {
+          "format": "int32",
+          "type": "integer"
+        },
+        "family": {
+          "type": "string"
+        },
+        "name": {
+          "type": "string"
+        },
+        "size": {
+          "type": "string"
+        },
+        "tier": {
+          "type": "string"
+        }
+      }
+    },
+    "ComputePolicyDto": {
+      "required": [
+        "name"
+      ],
+      "type": "object",
+      "allOf": [
+        {
+          "$ref": "#/definitions/ComputePolicyRequest"
+        }
+      ],
+      "properties": {
+        "id": {
```

Comment:
>id [](http://example.com/codeflow?start=9&length=2)

Don't duplicate top-level property names. If this is an arm resource Id, recommend using resourceId for the property name with format: arm-id. If it's some other kind of ID, please make it clear from the name and/or description what kind/format of ID it is.